### DISCUSSION TITLE
No wireless connection anymore - R.I.P. USB-RF Stick?

### DISCUSSION THREAD

**Discussion by ntdsutil** [OpenCCU/OpenCCU#2082] [🇩🇪 Fragen&Antworten] [answered]
Hi guys,

I'm now a bit lost and I want to raise my questions here.
**Setup**: RaspberryMatic within a proxmox instance and RF-USB Stick.

**Issue**: Wireless connection stops abruptly without any reason.
**Description**: It started with an error CarrierSense-level 100% which was raised 10 times. Since then it was not possible to reach any wireless components anymore.
from messages: 
`Dec  3 08:44:00 homematic-raspi user.err watchdog: DutyCycle-1F2995E6C0, NAME: 'CCU', TYPE: CCU2, CONNECTED: 1, DC: 3 %, CS: 100 %`

After rebooting the VM and plugging off and on the RF-USB the erros were gone but still no Communication possible.

I decided to restore several backups to check if any update or misconfiguration is the reason for this, but nothing helped.

Stick is properly detected: 
```
Dec  4 11:14:10 localhost kern.info kernel: [    1.744635] usb 3-1: Product: eQ-3 HmIP-RFUSB
Dec  4 11:14:10 localhost kern.info kernel: [   10.311195] usb 3-1: Found eQ-3 HmIP-RFUSB with serial 3014F711A0000XYZXYZXYZ at usb-0000:00:1d.0-1
Dec  4 11:14:10 localhost kern.info kernel: [   10.311423] raw-uart raw-uart: Registered new raw-uart device using underlying device eQ-3 HmIP-RFUSB@usb-0000:00:1d.0-1.
Dec  4 11:14:10 localhost kern.info kernel: [   10.315610] usbcore: registered new interface driver hb_rf_usb_2
```

But stil no communication possible. Can not reach any of the devices to fetch data. Sometimes 1-2 components react for a short period of time but I could only reproduce this once...

Also no CS error messages and Dutycyle is always 0% which wonders me...

I also rebooted the whole Proxmox center to check for any physical reasons (voltage on USB port), changed the USB port, changed the USB extension. 
I personally think the stick is gone - after 3 months - **but any help is appreciated.**

Is there a  way to check if the stick is working properly within the UI?
Can I easily buy a new stick and plug it in or is there anything to consider?

Best regards and thanks for helping me

Cheers

**Comment by ntdsutil** [comment]
Ich weiß echt nicht mehr weiter.
Ich hab heute weiterversucht und erstmal alle funkkomponenten vom strom getrennt.
Erst war dann die Verbindung zum DRAP gestört (Nicht erreichbar), dann habe ich diesen auch stromlos gemacht.
Danach haben sich erstmal alle Komponenten am BUS gemeldet, die bisher noch nicht eingelernt sind.

Kurz darauf ging der CS wieder von 0% auf 100% und ließ sich nicht wieder runterkriegen.

Wie wahrscheinlich ist es, dass eine Komponente auf dem BUS für Terror auf dem RF-Stick sorgt? Also nicht dass ich das ernsthaft vermute ich bin nur dermaßen verzweifelt^^

**Comment by ntdsutil** [comment]
Also ich habe jetzt die Sticks mal an einen Raspberry Pi mit frischer Installation angeschlossen. Dort zeigen die das gleiche Verhalten.
An meinen Komponenten liegt es nicht, da diese mit dem GPIO Modul einwandfrei gefunden wurden.

Kann es sein, dass beide Sticks defekt sind? Bzw. einer es von Anfang an war?
Gibt es Fehler beim löten ein solches Verhalten erklären können?

**Comment by MichaelN0815** [comment]
Ein zu hoher CS wird (höchstwahrscheinlich) nicht durch dein Funkmodul verursacht, sondern von diesem gemessen. Bei 100% CS ist keinen Kommunikation mehr möglich. Du musst erst den Störer beseitigen. Hat Dir Google keinen entsprechenden Hinweise ausgeworfen?  Dann hier lang: https://homematic-forum.de/forum/viewtopic.php?f=31&t=75532

**Comment by ntdsutil** [comment]
Hey @MichaelN0815!
Danke für deinen Kommentar und dein Interesse. Na klar, Google und sämtliche Foren schon durchstöbert...
Die Sache mit dem CS Wert konnte ich nur einmal reproduzieren als ich ihn förmlich produziert habe.
Im Normalfall habe ich 0% CS aber ich gebe zu, als das Ganze zusammengefallen ist, war das eine der Fehlermeldungen.

Warum ich Skepsis habe?
Beide Sticks an unterschiedlichen Orten sowohl innerhalb der Proxmox als auch am Raspi getestet --> gleiches Ergebnis
Gegentest mit Raspi und RF Funkmodul, keine Probleme.

Um das CS Thema endgültig auszuschließen: Wie zuverlässig ist der minütlich in /var/log/messages geschriebene CS Wert?

Danke und Gruß

**Reply by MichaelN0815** [reply]
Ich wusste gar nicht das der Wert ins Log geschrieben wird. Nun,  er ist insofern zuverlässig,  als er vom Funkmodul kommt. Und den Störpegel dokumentiert. Wenn CS 0 dann ist alles gut, ist der Wert höher gibt es Probleme. Genauer kann man es nicht sagen. Ist halt Funk. Und zum Gegentest: allein das verwendete Gerät, ob Rpi3 oder RPi4 macht schon einen riesen Unterschied.

**Reply by jens-maus** [reply]
> Um das CS Thema endgültig auszuschließen: Wie zuverlässig ist der minütlich in /var/log/messages geschriebene CS Wert?

Der vom Funkmodul bezogene CarrierSense Wert der bei Überschreitung auch im Logfile landet ist genau der Wert der vom Funkmodul bezogen wird. Und wenn dieses dauerhaft bzw. für einen längeren Zeitraum einen CS Wert > 10% liefert ist erfahrungsgemäß ein Störsender in der Umgebung des Funkmodules am Werk. Das kann ein Netzteil, ein Computer, ein Mikrowelle oder sonstiges elektrisches Gerät sein das eben das 868MHz Funkband nachhaltig stört und dazu führt das das Funkmodul meint das Funkband ist dicht und damit den Funkverkehr komplett einstellt bis der CS Wert wieder gesunken ist. Bis dahin ist dann keinerlei (vor allem kein HmIP traffic) möglich.

Oft hilft es bereits ein 2-3m langes USB Verlängerungskabel zu benutzen um das Funkmodul so weit wie möglich von irgendwelchen direkten Störquellen entfernt zu betreiben. Und auch noch zusätzlich einen USB-Entstörfilter (z.B. https://de.elv.com/elv-usb-entstoerfilter-usb-ef1-komplettbausatz-152745) einzusetzen hat sich als ratsam herausgestellt. Und natürlich sollte man ein gutes Auge auf den verlöteten Antennenstummel des HmIP-RFUSB haben und lieber doppelt kontrollieren das dieser korrekt verlötet und verlegt wurde. Darüberhinaus kann es natürlich auch sinnvoll sein statt diesem Stummen gleich auf eine externe Antenne zu setzen (Stichwort Groundplane-Antenne, z.B. https://homematic-forum.de/forum/viewtopic.php?f=27&t=27287) um diese dann auch soweit wie möglich weg zu positionieren und gleichzeitig noch die Sende-/Empfangsqualität zu erhöhen.

Aber wenn all das nichts bringt gibt es auch einige Beispiele von Nutzern die trotz aller Maßnahmen weiterhin hohe CS Werte hatten oder der Funkverkehr nicht oder nur ab und zu möglich war. Bei einigen Nutzern gab es sogar Beispiele das am Schluss an arglos weggeworfener pseudo-smarter Heizkörperthermostat  in der Mülltonne des Nachbarn samt montierter Batterien gefunden wurde und dieser einfach wie wild umsich gefunkt hat und damit das 868MHz Band in der näheren Nachbarschaft komplett dicht gemacht hat. Um das herauszufinden hilft es nur dann einen entsprechenden 868MHz SDR Scanner zur Hand zu haben um selbst auch die Suche nach dem Störsender zu gehen.... Funk ist eben Funk und das 868MHz Band ist auch nicht exklusiv für HomeMatic, sondern eine allgemeine, freie Frequenz auf der jeder (bis zum DutyCycle) funken kann was/wie er will. Und wenn sich da nicht jeder an den DutyCycle bzw. Gepflogenheiten hält passiert eben Beliebiges.

**Reply by ntdsutil** [reply]
Hey @jens-maus! 
Vielen Dank für deine ausführliche Antwort.
Primär nutze ich das virtualisierte Setup mit USB Stick. Der Raspi ist nur Versuchsaufbau.
Ich checke nachher nochmal alle Lötstellen in beiden Sticks um das als Fehler auszuschließen. Wäre das ein potentieller Fehler? Wenn bspw. das Funkmodul nicht korrekt verlötet ist, dann würde das nicht vom System erkannt? Kann man einen spezifischen RSSI irgendwo dem Log entnehmen?

Die Geschichten zu den Störern habe ich mir alle durchgelesen, abgefahren wie leicht das Signal zu stören ist. Vor allem bei mir lief alles 3 Monate so geschmeidig und da waren sicher mehr als genug Störer in der Nähe. Meine letzten Test fanden in einem Kellerraum ohne potentielle Störquellen statt und wie gesagt in der Regel habe ich 0% CS...

DANKE

**Comment by ntdsutil** [comment, marked as answer]
Ok, also ich hatte ja aufgegeben und einfach den dritten Stick bestellt und siehe da, es geht wieder.
Sehr schade, dass ein Stick nach 3 Monaten gegangen ist und ein zweiter direkt schon kaputt geliefert wurde. Werde diesen mal über Weihnachten durchmessen und schauen wo es klemmt. Danke an alle hier <3

**Comment by TheJokerHA** [comment]
Hi,

habe so ein ähnliches Problem, ob der Stick defekt ist kann ich mir schlecht vorstellen, lief ja letzte Woche noch und alles wird erkannt!!!

![1](https://user-images.githubusercontent.com/85641621/207299646-efd136f3-ec3b-4298-a996-01c3564c9b31.JPG)
![2](https://user-images.githubusercontent.com/85641621/207299911-f9061f0a-30e2-45f4-bbb6-6c8c06cf3830.JPG)

Komisch ist das im Log file die Info steht!

ec 13 10:10:01 de838cd8-raspberrymatic user.warn monit[1255]: 'hasUSB' status failed (1) -- no output
Dec 13 10:10:02 de838cd8-raspberrymatic user.err monit[1255]: 'rpi4usb3Check' status failed (1) -- no output
Dec 13 10:10:02 de838cd8-raspberrymatic user.err monit[1255]: Filesystem '/media/usb1' not mounted
Dec 13 10:10:02 de838cd8-raspberrymatic user.err monit[1255]: 'usb1' unable to read filesystem '/media/usb1' state
Dec 13 10:10:02 de838cd8-raspberrymatic user.info monit[1255]: 'usb1' trying to restart

[de838cd8-raspberrymatic-2022-12-13.log](https://github.com/jens-maus/RaspberryMatic/files/10217166/de838cd8-raspberrymatic-2022-12-13.log)

Hat jemand eine Idee warum der Fehler jetzt auftritt bzw. keine Verbindung besteht ?? Bevor ich jetzt eine neuen Stick bestelle ??.

Danke für eure Hilfe :-)

**Reply by jens-maus** [reply]
> habe so ein ähnliches Problem, ob der Stick defekt ist kann ich mir schlecht vorstellen, lief ja letzte Woche noch und alles wird erkannt!!!
[...] 
> Hat jemand eine Idee warum der Fehler jetzt auftritt bzw. keine Verbindung besteht ?? Bevor ich jetzt eine neuen Stick bestelle ??.

Was denn für einen Fehler? In deinen Ausgaben und im Screenshot sehe ich keinerlei Fehler. das HmIP-RFUSB wird doch korrekt erkannt (überall OK im boot log) und auch in der Infoseite werden die Adresse und die Seriennummer korrekt ausgegeben. Und auch deine monit ausgaben hier zeigen keinerlei Fehler auf bzgl. des Funkmodules.

Wenn du allerdings trotz dieses fehlerlosen Status keine Geräte anlernen oder erreichen kannst liegt dein problem eben auf einer anderen Ebene. Entweder zu starke Interferenzen in der näheren Umgebung oder das Funkband ist dicht oder oder oder. Aber an RaspberryMatic bzw. der Erkennung des HmIP-RFUSB liegt es definitiv nicht. Das sieht alles OK aus was du hier präsentiert hast.

**Comment by TheJokerHA** [comment]
Hi,

das ging ja schnell :-) mit einer Antwort.

Problem ist, das keine Kommunikation zu der Teilnehmern mehr besteht!

![3](https://user-images.githubusercontent.com/85641621/207304191-72c11cac-e327-4ecd-a26a-1455b03c8776.JPG)

Was bedeutet den diese Info den ??

ec 13 10:10:01 de838cd8-raspberrymatic user.warn monit[1255]: 'hasUSB' status failed (1) -- no output
Dec 13 10:10:02 de838cd8-raspberrymatic user.err monit[1255]: 'rpi4usb3Check' status failed (1) -- no output
Dec 13 10:10:02 de838cd8-raspberrymatic user.err monit[1255]: Filesystem '/media/usb1' not mounted
Dec 13 10:10:02 de838cd8-raspberrymatic user.err monit[1255]: 'usb1' unable to read filesystem '/media/usb1' state
Dec 13 10:10:02 de838cd8-raspberrymatic user.info monit[1255]: 'usb1' trying to restart

Das mit den Interferenzen werde ich mal testen, längeres Kabel usw... 
Kann es sein das wie bei meinem Vorredner das Modul defekt ist??

**Reply by jens-maus** [reply]
> Problem ist, das keine Kommunikation zu der Teilnehmern mehr besteht!
[...]

Das ist doch ne glasklare Sache anhand des Screenshots: Du hast einen Funkstörer in der nähe des Funkmodules (anderes technisches Gerät, Mikrowelle, Funkaktor der am Ende der Batterielaufzeit ist und wild rumfunkt, und und und).

Da kann dir auch kein anderer helfen. Da musst du schon selbst auf die Suche gehen und den Störsender suchen (1. in deiner eigenen Umgebung durch abschalten von Strom bzw. rausnahme von batterien, etc. etc. oder wenn das alles nix bringt in der Nachbarschaft). 868MHz ist ja eine geteile Funkfrequenz und da kann es eben wie mit WLANs irgendwie eng werden.

> Was bedeutet den diese Info den ??

Nichts was darauf schließen lässt das das Funkmodul ein Problem hätte. Das sind Meldungen dazu das du keinen USB Memorystick angeschlossen hast. Mehr nicht.

> Das mit den Interferenzen werde ich mal testen, längeres Kabel usw... Kann es sein das wie bei meinem Vorredner das Modul defekt ist??

Nein. Wenn das Funkmodul defekt ist dann wird es komplett nicht erkannt. Und ehrlich gesagt glaube ich auch nicht recht an ein defektes Funkmodul beim Vorredner. Das wird alles nur Zufall sein oder ähnliches. Funk ist eben Funk und da kann und wir immer mal was Schiefgehen. Nur wer auf Kabelverdrahtung setzt ist störfrei. Sei froh das dir das Funkmodul wenigstens anzeigt das da irgendwo ein Störsender ist auf der Frequenz. Ohne diese Info würdest du komplett im Trüben fischen. Nun liegt es an dir selbst den Störsender zu finden!

**Reply by ntdsutil** [reply]
Hi! Also ich kann mir für mich sprechen. Ob das Zufall war oder nicht kann mir keiner sagen. Seitdem ich aber die lötstellen von Stick 2 erneuert habe, funktioniert auch der einwandfrei. Ich hätte diese Meldung zum Carrier sense nur 2 Mal und da habe ich diese förmlich provoziert. Ich war ja relativ verzweifelt, da irgendwie nichts geholfen hat, und habe sämtliche Funkkomponenten vom Strom genommen. Die gesamte Umgebung vom Strom getrennt und physikalisch an einen Ort umgezogen und dort weitergetestet. Habe sogar einen zweiten Aufbau mit raspi und gpio Modul genutzt um die Komponenten zu prüfen. Erst als ich den dritten Stick dann erfolgreich am raspi getestet habe bin ich wieder zurück in meine ursprüngliche Umgebung und seitdem geht auch wieder alles. Kann natürlich sein, dass durch den kompletten shutdown ein störer eliminiert wurde aber das würde mir weder angezeigt (keine CS Meldung) noch hat das herunterfahren in den ersten Tagen Erfolg gehabt. Wie gesagt kann aber auch echt blöder Zufall sein und morgen steh ich wieder dumm da aber bisher läufts

**Reply by MichaelN0815** [reply]
Ganz heißer Tip: wenn da Anzeigen in ROT sind, dann hat das meist was zu bedeuten... Die Alarm Meldung weist wahrscheinlich auch auf den CS hin. Was soll Jens denn noch tun, damit der Anwender die richtigen Infos bekommt?

**Comment by Stephan1908** [comment]
Hallo zusammen,

ich habe auch ein seltsames verhalten, nach einem Stromausfall werden keine Geräte mehr erkannt, unter Einstellungen=>Geräte habe ich keine RSSI Werte mehr:

![image](https://github.com/jens-maus/RaspberryMatic/assets/60405276/17eb418c-23f5-49b2-a93e-ef43e08f91f6)

Der Log nach dem letzten Neustart sieht so aus:

![image](https://github.com/jens-maus/RaspberryMatic/assets/60405276/57155df4-c93e-411b-afa9-6023741bd018)

Und mein Duty Cycle und Carrier Sense so:

![image](https://github.com/jens-maus/RaspberryMatic/assets/60405276/9cab07fb-a41b-4ebb-af99-1dbb570fa864)

Unter Hilfe müsste der HMIP-RFUSB ja richtig erkannt worden sein?

![image](https://github.com/jens-maus/RaspberryMatic/assets/60405276/1e28b39e-1442-4490-a532-6d65ece1eafa)


Hat noch jemand eine Idee, was ich prüfen könnte, oder woran es liegen könnte?

Vielen Dank

Grüße Stephan
